src/cpp/session/modules/SessionProfiler.cpp

```diff
@@ -65,10 +65,40 @@ void handleProfilerResReq(const http::Request& request,
    pResponse->setCacheableFile(profileResource, request);
 }
 
+void onDocPendingRemove(const std::string &id)
+{
+   // retrieve document from source database
+   boost::shared_ptr<source_database::SourceDocument> pDoc(
+               new source_database::SourceDocument());
+   Error error = source_database::get(id, pDoc);
+   if (error)
+   {
+      LOG_ERROR(error);
+         return;
+   }
+
+   // check to see if there is html cached data
+   std::string htmlLocalPath = pDoc->getProperty("htmlLocalPath");
+   if (htmlLocalPath.empty())
+      return;
+
+   r::sexp::Protect rProtect;
+   r::exec::RFunction rFunction(".rs.rpc.clear_profile");
+   rFunction.addParam(htmlLocalPath);
+
+   SEXP resultSEXP;
+   error = rFunction.call(&resultSEXP, &rProtect);
```

Comment:
I don't think the dummy result/protect vars are needed here (you can just `call()` if you don't care about the return value).